dir.props

```diff
@@ -436,8 +436,8 @@
     <PortableNuGetMode>true</PortableNuGetMode>
   </PropertyGroup>
 
-  <PropertyGroup>
-    <!-- Never use the NuGet fallback folder that comes with the SDK we use to build.
+  <PropertyGroup Condition="'$(MSBuildProjectName)' != 'crossgen'">
```

Comment:
?? I'm not sure this is the right thing to do.  We should NEVER be using the fallback folder.

Reply:
The crossgen build is part of buildtools. When restoring buildtools we will build crossgen against the same runtime that will be used to run msbuild in order to crossgen the whole repo. For that to work, we need to be able to restore the packages for this crossgen project comming from buildtools, and the dependencies that we need for this new version of the CLI are not in myget, so that is why we just use the fallback folder for this project only, since it will contain the missing packages.

Reply:
Can't we just set this property in that project then?  Or even better - crossgen.csproj shouldn't be importing the repo's dir.props.  The dir.props is only for projects that are actually in this repo.

---
In reply to: [165137167](https://github.com/dotnet/core-setup/pull/3656#discussion_r165137167) [](ancestors = 165137167)

Reply:
crossgen is an SDK project, and given that there is a Directory.Build.props on core-setup, then dir.props get's loaded automatically, and there is no easy way of disabling this other than making it not be a SDK project which I believe is not what we want. I will explore that suggestion of trying to instead set the property manually, but given that dir.props gets imported automatically, it might be imported late.

Reply:
`dir.props` will get imported first and set the property to `true`.  Then your project will set the property and overwrite it to `false`.  As the project, you always have full control to set things exactly as you want.

Reply:
I agree we should not special case crossgen project here but instead set the property in that project.

Reply:
I tried that locally and it does seem to work so I'll make the change in buildtools and update this PR.